src/main/java/uk/gov/hmcts/reform/preapi/services/BookingService.java

```diff
@@ -97,4 +107,23 @@ public void markAsDeleted(UUID id) {
             bookingRepository.deleteById(id);
         }
     }
+
+    @Transactional
+    public UpsertResult shareBookingById(ShareBookingDTO shareBookingDTO) {
+        var booking = bookingRepository.findById(shareBookingDTO.getBookingId())
+            .orElseThrow(() -> new NotFoundException("Booking: " + shareBookingDTO.getBookingId()));
+        var sharedByUser = userRepository.findById(shareBookingDTO.getSharedByUserId())
+            .orElseThrow(() -> new NotFoundException("Shared by User: " + shareBookingDTO.getSharedByUserId()));
+        var sharedWithUser = userRepository.findById(shareBookingDTO.getSharedWithUserId())
+            .orElseThrow(() -> new NotFoundException("Shared with User: " + shareBookingDTO.getSharedWithUserId()));
+
+        var shareBookingEntity = new ShareBooking();
+        shareBookingEntity.setId(shareBookingDTO.getId());
+        shareBookingEntity.setBooking(booking);
+        shareBookingEntity.setSharedBy(sharedByUser);
+        shareBookingEntity.setSharedWith(sharedWithUser);
+        shareBookingRepository.save(shareBookingEntity);
+
+        return UpsertResult.CREATED;
```

Comment:
Are we only allowing create on this endpoint? It makes sense but if we are we probably want an extra check to make sure we aren't accidentally updating.
```java
if(shareBookingRepository.existsById(shareBookingDTO.getId()) {
    // throw an error
}
```